### PR TITLE
Add GraphQL subscription support

### LINKED ISSUE #237: Support graphql's subscription

### PR DESCRIPTION
**Pull request by spitko**
Resolves #237 

## Node package

Added trigger function send_notification() that sends notification using pg_notify to topic [schema].[name]

Notification payload (JSON string) consists of:
1. id (Entity ID),
2. mutation_type (INSERT/UPDATE/DELETE)
3. _entity (JSON-formatted entity)

Note that pg_notify has a payload size limit of 8000 bytes

## Query package

- Used pg-pubsub plugin to handle listening to Postgres notifications 
- Created a custom graphile plugin to add subscription fields to schema
  - Entity field is currently typed as JSON
- Used subscriptions-transport-ws for websocket server
  - this should be replaced with graphql-ws once [support is added to playground](https://github.com/graphql/graphql-playground/issues/1291)

### CODE REVIEW

**Comment by ianhe8x** [comment]
Thanks for contribution, we will look at it as soon as we can

File: packages/node/src/utils/sync-helper.ts [hunk 1]
```diff
@@ -37,3 +37,34 @@ export function createUniqueIndexQuery(
...
+        'mutation_type', TG_OP,
+        '_entity', row)::text);
+    RETURN NULL;
+END;
+$$ LANGUAGE plpgsql;`;
+
+export function createNotifyTrigger(schema: string, table: string): string {
+  return `
+DROP TRIGGER IF EXISTS ${schema}_${table}_notify_trigger
+    ON ${schema}.${table};
```
Comment (jiqiang90) [inline packages/node/src/utils/sync-helper.ts:65]: suggest wrap `${schema}` with double quote "" 

if we have project name in camelCase, like "dragonFly", when init the project your will get error like : 

```
<store> ERROR Having a problem when syncing schema SequelizeDatabaseError: schema "dragonfly" does not exist

// The query

DROP TRIGGER IF EXISTS dragonFly_starter_entities_notify_trigger
    ON dragonFly.starter_entities;
CREATE TRIGGER dragonFly_starter_entities_notify_trigger
    AFTER INSERT OR UPDATE OR DELETE
    ON dragonFly.starter_entities
    FOR EACH ROW EXECUTE FUNCTION send_notification(); 

```

We can fix it with double quote

```
DROP TRIGGER IF EXISTS dragonFly_starter_entities_notify_trigger
    ON "dragonFly".starter_entities;

```

File: packages/node/src/utils/sync-helper.ts [hunk 2]
```diff
@@ -37,3 +37,34 @@ export function createUniqueIndexQuery(
...
+CREATE TRIGGER ${schema}_${table}_notify_trigger
+    AFTER INSERT OR UPDATE OR DELETE
+    ON ${schema}.${table}
```
Comment (jiqiang90) [inline packages/node/src/utils/sync-helper.ts:68]: add double quote here

Comment (jiqiang90) [reply packages/node/src/utils/sync-helper.ts:68]: Hi @spitko,  also can we add this feature under `--unsafe` mode ? Thank you 

Other LGTM

**Comment by spitko** [comment]
Thanks for review @jiqiang90, just pushed fixes

**Comment by spitko** [comment]
Pushed a fix to set entity as null in notification payload if payload size is over the 8000 byte limit

**Comment by spitko** [comment]
Pushed a fix to set entity as null in notification payload if payload size is over the 8000 byte limit

**Comment by jiqiang90** [comment]
Thanks @spitko, one more thing, can we add filter for id and mutation_type?

**Comment by spitko** [comment]
> Thanks @spitko, one more thing, can we add filter for `id` and `mutation_type`?

Done

**Comment by jamesbayly** [comment]
@spitko  I strongly suggest you also take a look at the [Historical Query Support](https://gitcoin.co/issue/subquery/grants/7/100028529)

**Review by ianhe8x — APPROVED** [review]

**Pull request merged**